### PR TITLE
added basket item component

### PR DESCRIPTION
**Pull request by robhuzzey**
#### What does this PR do? (please provide any background)
Introduces a simple basket item component to the library.

#### What tests does this PR have?
Unit tests that can be run via `npm run test`

#### How can this be tested?
I would run `npm run test` to see unit test passing then run `npm run docs` & see it for yourself on the demo page.

#### Screenshots / Screencast
N/A

#### What gif best describes how you feel about this work?
battles

#### Developer Definition of Done/Quality Checklist (for PR author to complete BEFORE code review):
- [x] I have checked the [contributing document](../blob/master/CONTRIBUTING.md) and I'm happy for this to be reviewed.

#### Software Engineer or Developer review:
- [x] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [x] I’ve checked for coding anti-patterns.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked all the tests are passing.

#### Software Engineer or project guru review:
- [x] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [x] I’ve checked for coding anti-patterns.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked all the tests are passing.

### CODE REVIEW

**Comment by jodiedoubleday** [comment]
:fist: Taking SE

File: docs/examples/BasketItem.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+var example = (
+  <div className="panel">
```
Comment (jodiedoubleday) [inline docs/examples/BasketItem.jsx]: I probably wouldn't add the panel `className` here. It creates unnecessary styling for the docs, and is a bootstrap class. Although the docs uses bootstrap it isn't necessary for the demo.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ui-toolkit",
-  "version": "0.18.0",
+  "version": "0.19.0",
```
Comment (jodiedoubleday) [inline package.json]: no need to change this, this is updated automatically when you release. If you change this here I'm pretty sure it will skip a version.

File: src/components/basket-item/_basket-item.less [hunk 3]
```diff
@@ -0,0 +1,20 @@
+/*------------------------------------*\
+    #Basket Item
+\*------------------------------------*/
+
+
+.component-basket-item {
```
Comment (jodiedoubleday) [inline src/components/basket-item/_basket-item.less]: We need a fallback here for older browsers.

File: src/components/basket-item/_basket-item.less [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  display: flex;
```
Comment (jodiedoubleday) [inline src/components/basket-item/_basket-item.less]: before `display:flex` do `display:table` and `table-layout:fixed`

File: src/components/basket-item/_basket-item.less [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  flex-direction: row;
+  flex-wrap: nowrap;
+  justify-content: flex-start;
+  align-items: flex-start;
+  align-content: stretch;
+
+  .component-basket-item-description {
```
Comment (jodiedoubleday) [inline src/components/basket-item/_basket-item.less]: add `display:table-cell` on this element

File: src/components/basket-item/_basket-item.less [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    flex: 1;
+  }
+  .component-basket-item-price {
```
Comment (jodiedoubleday) [inline src/components/basket-item/_basket-item.less]: add `display:table-cell` on this element

File: test/components/basket-item-test.jsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+      it('should return the price with the correct currency', function() {
+        var basketItem = TestUtils.renderIntoDocument(<BasketItem price={123}>I am content</BasketItem>);
+        var renderedBasketItem = TestUtils.findRenderedDOMComponentWithClass(basketItem, 'component-basket-item-price');
+        assert.equal(renderedBasketItem.getDOMNode().textContent, '£123');
+      });
+    });
+
+    describe('with a currency symbol passed in', function() {
+      it('should return the price with the passed in currency symbol', function() {
+        var basketItem = TestUtils.renderIntoDocument(<BasketItem price={456} currencySymbol="€">I am content</BasketItem>);
```
Comment (jodiedoubleday) [inline test/components/basket-item-test.jsx]: I don't want to be prescriptive about currencies, but can i type anything into the currency field. Like the new currency of Jodie Bucks `ȷ324`

File: test/components/countdown-test.js [hunk 8]
```diff
@@ -165,7 +165,7 @@ describe('Countdown tests', function() {
       });
 
       it('should return 10 days', function() {
-        assert.equal(duration.asDays(), -10);
+        assert.equal(Math.ceil(duration.asDays()), -10);
```
Comment (jodiedoubleday) [inline test/components/countdown-test.js]: thank god you have fixed this, should this be in a different PR?

Comment (jodiedoubleday) [inline test/components/countdown-test.js]: Why is this file no in the tests folder with the rest of the tests?

File: test/components/basket-item-test.jsx (hunk 7 above)
Comment (robhuzzey) [inline test/components/basket-item-test.jsx]: You could... this is kinda why I left it open & called it "currencySymbol" rather than "currency". I can see us in the future having `currency="GBP"` on a pricing node or something & have that deal with the symbol... but for now... you can use "jodieBucks" if you wish ;)

File: test/components/countdown-test.js (hunk 8 above)
Comment (robhuzzey) [inline test/components/countdown-test.js]: Considered it... figured it's a quick fix so left it in here.

**Comment by jodiedoubleday** [comment]
All good, dare I say can you test this in IE. (10 or 11)

File: test/components/countdown-test.js (hunk 8 above)
Comment (hxpaul) [inline test/components/countdown-test.js]: this is not the right fix for this imo, if it started returning 9.5 or 9.00001 we'd not know... we just need to cover the clocks going back, which you can do with this:
```
   describe('_durationFromNow', function() {
     var clock;
    beforeEach( function( ) {
      clock = sinon.useFakeTimers( new Date( 2015, 1, 1 ).getTime( ));
    } );
    afterEach( function( ) {
      return clock.restore( );
    } );
```

**Comment by hxpaul** [comment]
it feels like a basket item should have more to it, at least the link to remove something from the basket. Then that "click here to remove this" can be standardised and all you'd have to do is pass in a `removeHandler`
apart from that and the way you have fixed the dodgy countdown test, all looks good, and good work on updating the linting, I know making those changes makes reviewing your pr offputting.

**Comment by robhuzzey** [comment]
@hxpaul I may iterate on this as I agree with you... I wanted to start simple first & see how it's used before doing more than may/may not be necessary.

@jodiedoubleday can I get some ticks on this please & @hxpaul is you are 2nd reviewer, some from you too please?

**Comment by hxpaul** [comment]
I can tick them when you change that countdown test

**Comment by hxpaul** [comment]
looking good here

**Comment by robhuzzey** [comment]
@jodiedoubleday just need your ticks on this please :)
